db/migrate/20220521032400_create_canonical_recipes_ingredients.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class CreateCanonicalRecipesIngredients < ActiveRecord::Migration[6.1]
+  def change
+    create_table :canonical_recipes_ingredients do |t|
+      t.references :recipe, null: false, foreign_key: { to_table: 'canonical_books' }
```

Comment:
It goes against the Rails Way™️ a bit but I decided to name the foreign key here `recipe_id` because the only books that can be associated to this join table are recipes.